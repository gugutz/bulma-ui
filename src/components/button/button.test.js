import React from 'react'
import renderer from 'react-test-renderer'
import Button from './button'

describe('<Button /> component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Button>text</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support type prop', () => {
    const tree = renderer.create(<Button type="submit">text</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  const colors = [
    'white',
    'light',
    'dark',
    'black',
    'text',
    'primary',
    'link',
    'info',
    'success',
    'warning',
    'danger'
  ]

  for (const color of colors) {
    it('should support color prop with ' + color, () => {
      const tree = renderer.create(<Button color={color}>text</Button>).toJSON()
      expect(tree).toMatchSnapshot()
    })
  }

  const sizes = ['small', 'normal', 'medium', 'large']

  for (const size of sizes) {
    it('should support size prop with ' + size, () => {
      const tree = renderer.create(<Button size={size}>text</Button>).toJSON()
      expect(tree).toMatchSnapshot()
    })
  }

  it('should support isOutlined prop', () => {
    const tree = renderer.create(<Button isOutlined>text</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support isInverted prop', () => {
    const tree = renderer.create(<Button isInverted>text</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support isRounded prop', () => {
    const tree = renderer.create(<Button isRounded>text</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  const states = ['normal', 'hovered', 'focused', 'active', 'loading']

  for (const state of states) {
    it('should support state prop with ' + state, () => {
      const tree = renderer.create(<Button state={state}>text</Button>).toJSON()
      expect(tree).toMatchSnapshot()
    })
  }

  it('should support isStatic prop', () => {
    const tree = renderer.create(<Button isStatic>text</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support isSelected prop', () => {
    const tree = renderer.create(<Button isSelected>text</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support isFullWidth prop', () => {
    const tree = renderer.create(<Button isFullWidth>text</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
