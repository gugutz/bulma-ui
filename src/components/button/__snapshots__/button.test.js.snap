// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> component should render correctly 1`] = `
<button
  className="button"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support color prop with black 1`] = `
<button
  className="button is-black"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support color prop with danger 1`] = `
<button
  className="button is-danger"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support color prop with dark 1`] = `
<button
  className="button is-dark"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support color prop with info 1`] = `
<button
  className="button is-info"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support color prop with light 1`] = `
<button
  className="button is-light"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support color prop with link 1`] = `
<button
  className="button is-link"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support color prop with primary 1`] = `
<button
  className="button is-primary"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support color prop with success 1`] = `
<button
  className="button is-success"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support color prop with text 1`] = `
<button
  className="button is-text"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support color prop with warning 1`] = `
<button
  className="button is-warning"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support color prop with white 1`] = `
<button
  className="button is-white"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support isInverted prop 1`] = `
<button
  className="button is-inverted"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support isOutlined prop 1`] = `
<button
  className="button is-outlined"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support isRounded prop 1`] = `
<button
  className="button is-rounded"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support isSelected prop 1`] = `
<button
  className="button is-selected"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support isStatic prop 1`] = `
<button
  className="button is-static"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support size prop with large 1`] = `
<button
  className="button is-large"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support size prop with medium 1`] = `
<button
  className="button is-medium"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support size prop with normal 1`] = `
<button
  className="button"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support size prop with small 1`] = `
<button
  className="button is-small"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support state prop with active 1`] = `
<button
  className="button is-active"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support state prop with focused 1`] = `
<button
  className="button is-focused"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support state prop with hovered 1`] = `
<button
  className="button is-hovered"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support state prop with loading 1`] = `
<button
  className="button is-loading"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support state prop with normal 1`] = `
<button
  className="button"
  disabled={false}
  type="button"
>
  text
</button>
`;

exports[`<Button /> component should support type prop 1`] = `
<button
  className="button"
  disabled={false}
  type="submit"
>
  text
</button>
`;
